File: apps/mobile/test/roles_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:servis_takip/core/constants/roles.dart';

void main() {
  test('her rol için bir ana ekran rotası tanımlıdır', () {
    for (final role in Role.values) {
      expect(role.homeRoute.isNotEmpty, isTrue);
    }
  });

  test('operations_manager ve company_admin aynı yönetici arayüzünü paylaşır', () {
    expect(Role.operationsManager.homeRoute, Role.companyAdmin.homeRoute);
  });

  test('fromValue geçerli/geçersiz değerleri ayırır', () {
    expect(Role.fromValue('driver'), Role.driver);
    expect(Role.fromValue('super_admin'), Role.superAdmin);
    // Bilinmeyen değer güvenli varsayılana (passenger) düşer.
    expect(Role.fromValue('hacker'), Role.passenger);
  });
}

File: apps/mobile-legacy-rn/src/__tests__/roles.test.ts
import { ROLES, ROLE_HOME_SEGMENT, isRole, ALL_ROLES } from '@servis/shared';

describe('rol tanımları', () => {
  it('her rol için bir ana ekran segmenti tanımlıdır', () => {
    for (const role of ALL_ROLES) {
      expect(ROLE_HOME_SEGMENT[role]).toBeTruthy();
    }
  });

  it('operations_manager ve company_admin aynı yönetici arayüzünü paylaşır', () => {
    expect(ROLE_HOME_SEGMENT[ROLES.OPERATIONS_MANAGER]).toBe(ROLE_HOME_SEGMENT[ROLES.COMPANY_ADMIN]);
  });

  it('isRole geçerli/geçersiz değerleri ayırır', () => {
    expect(isRole('driver')).toBe(true);
    expect(isRole('hacker')).toBe(false);
    expect(isRole(123)).toBe(false);
  });
});
